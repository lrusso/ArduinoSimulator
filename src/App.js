import React from "react"
import Editor from "./screens/Editor"
import { declareLanguageData } from "./utils/service"

const App = () => {
  const [webLoaded, setWebLoaded] = React.useState(false)

  React.useEffect(() => {
    declareLanguageData()

    const checkAceEditor = setInterval(() => {
      if (global.ace) {
        setWebLoaded(true)
        clearInterval(checkAceEditor)
      }
    }, 200)
  }, [])

  if (!webLoaded) {
    return null
  }

  return <Editor />
}

export default App
