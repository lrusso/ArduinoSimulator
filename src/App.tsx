import React from "react"
import Editor from "./screens/Editor"
import { declareLanguageData } from "./utils/languages"

import "./arduino-simulator.v1.0.0.css";

const App = () => {
  const [webLoaded, setWebLoaded] = React.useState<boolean>(false)

  React.useEffect(() => {
    declareLanguageData()

    const checkAceEditor = setInterval(() => {
      if (global.ace) {
        setWebLoaded(true)
        clearInterval(checkAceEditor)
      }
    }, 200)
  }, [])

  if (!webLoaded) {
    return null
  }

  return <Editor />
}

export default App
