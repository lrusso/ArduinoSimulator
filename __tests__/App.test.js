import React from "react"
import renderer from "react-test-renderer"
import App from "../src/App"

jest.mock("../src/utils/service", () => {
  return {
    ...jest.requireActual("../src/utils/service"),
    editorInit: jest.fn(),
  }
})

describe("App", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<App />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
