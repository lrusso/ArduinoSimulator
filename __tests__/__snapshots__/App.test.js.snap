// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering 1 1`] = `null`;
